### DISCUSSION TITLE
Are the sources for the demo app available?

### DISCUSSION THREAD

**Discussion by 1001QAdotNET** [Textualize/textual#1754] [Q&A] [answered]
I am asking about the sources for this app
https://github.com/Textualize/textual#about

**Comment by davep** [comment, marked as answer]
Aye, they are: https://github.com/Textualize/textual/blob/main/src/textual/demo.py

**Reply by 1001QAdotNET** [reply]
Amazing.. thank you for the quick reply!

**Reply by davep** [reply]
Most welcome. :-)

**Reply by 1001QAdotNET** [reply]
On Python 3.10.6 I installed pip install textual[dev} and then I tried to run the app
I am getting an `importError Package textual.widget has no class Switch`

That is line 62 in textual/widgets/__init__.py

**Reply by davep** [reply]
You're likely mixing the code from `main` with the latest released version of Textual installed from PyPi. If you want to use the version of `demo.py` that is from the last released version and you're downloading directly from the repo [it's going to be best to use the v0.10.1-tagged version](https://github.com/Textualize/textual/blob/v0.10.1/src/textual/demo.py).

`main` isn't always the latest release; it's almost always ahead of the latest release.

**Reply by 1001QAdotNET** [reply]
Got it thank you
I had to also download the css file and after that it worked
QQ: are the tables in the table widget editable ?

**Reply by 1001QAdotNET** [reply]
I can see in the docs that the widget is desinged to dispaly not to edit

**Reply by davep** [reply]
If you mean the `DataTable`, yes, in the currently read-only; but in the next release of Textual the API will be such that you'll be able to update values within it.

**Reply by 1001QAdotNET** [reply]
That is very good timing for me, looking forward to it, thank you for this
